src/linker/configuration.py

```diff
@@ -210,22 +168,50 @@ def _get_required_attribute(environment: Dict[Any, Any], key: str) -> str:
         return value
 
     @staticmethod
-    def _get_requests(environment: Dict[Any, Any], key: str) -> Dict[Any, Any]:
+    def _get_requests(
```

Comment:
It doesn't seem like there's any overlap between the case where the `key` is "spark" and the case where it's not. That suggests they should just be separate methods.